src/cross_section.F90

```diff
@@ -133,107 +140,142 @@ end subroutine calculate_xs
 ! given index in the nuclides array at the energy of the given particle
 !===============================================================================
 
-  subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, i_log_union)
+  subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, &
+       i_log_union, sqrtkT)
     integer, intent(in) :: i_nuclide ! index into nuclides array
     integer, intent(in) :: i_sab     ! index into sab_tables array
     real(8), intent(in) :: E         ! energy
     integer, intent(in) :: i_mat     ! index into materials array
     integer, intent(in) :: i_nuc_mat ! index into nuclides array for a material
     integer, intent(in) :: i_log_union ! index into logarithmic mapping array or
                                        ! material union energy grid
+    real(8), intent(in) :: sqrtkT    ! Square root of kT, material dependent
 
     integer :: i_grid ! index on nuclide energy grid
     integer :: i_low  ! lower logarithmic mapping index
     integer :: i_high ! upper logarithmic mapping index
     real(8) :: f      ! interp factor on nuclide energy grid
+    real(8) :: sigT, sigA, sigF ! Intermediate multipole variables
     type(Nuclide), pointer :: nuc
-    type(Material),   pointer :: mat
+    type(Material), pointer :: mat
 
     ! Set pointer to nuclide and material
     nuc => nuclides(i_nuclide)
     mat => materials(i_mat)
 
-    ! Determine index on nuclide energy grid
-    select case (grid_method)
-    case (GRID_MAT_UNION)
+    ! If MP, don't interpolate, it's all already baked in.
+    if (nuc % mp_present .and. &
+         (E >= nuc % multipole % start_E/1.0e6_8 .and.&
+          E <= nuc % multipole % end_E/1.0e6_8)) then
```

Comment:
Can we break this into two `if`s (one to check `mp_present` and another nested one to check energies)? Fortran doesn't guarantee short-circuit evaluation so this is not safe if no multipole data is present.